APMrover2/GCS_Mavlink.cpp

```diff
@@ -640,6 +640,42 @@ void GCS_MAVLINK_Rover::handle_change_alt_request(AP_Mission::Mission_Command &c
     // nothing to do
 }
 
+MAV_RESULT GCS_MAVLINK_Rover::handle_command_do_set_mode(const mavlink_command_long_t &packet)
```

Comment:
I think we should remove support for the set mode that uses the MAV_MODE_* enums. They are just misleading. No real GCS would send this type of mode set.